odl/space/fspace.py

```diff
@@ -359,192 +481,416 @@ def element(self, fcall=None, fapply=None):
             If fcall is a `FunctionSetVector`, it is wrapped
             as a new `FunctionSpaceVector`.
 
-        fapply : `callable`, optional
-            The actual instruction for in-place evaluation.
-            Its first argument must be the `FunctionSet.range` element
-            or the array of such (vectorization) to which the
-            result is written.
-
-            If fapply is a `FunctionSetVector`, it is wrapped
-            as a new `FunctionSpaceVector`.
+        vectorized : bool
+            Whether ``fcall`` supports vectorized evaluation.
 
         Returns
         -------
         element : `FunctionSpaceVector`
-            The new element.
+            The new element, always supports vectorization
+
+        Notes
+        -----
+        If you specify ``vectorized=False``, the function is decorated
+        with a vectorizer, which makes two elements created this way
+        from the same function being regarded as *not equal*.
         """
-        if fcall is None and fapply is None:
+        if fcall is None:
             return self.zero()
         else:
-            return super().element(fcall, fapply)
+            if not callable(fcall):
+                raise TypeError('function {!r} is not callable.'.format(fcall))
+            if not vectorized:
+                if self.field == RealNumbers():
+                    dtype = 'float64'
+                else:
+                    dtype = 'complex128'
+
+                fcall = vectorize(dtype=dtype)(fcall)
+
+            return self.element_type(self, fcall)
+
+    def zero(self):
+        """The function mapping everything to zero.
+
+        This function is the additive unit in the function space.
+
+        Since `lincomb` may be slow, we implement this function
+        directly.
+        """
+        dtype = complex if self.field == ComplexNumbers() else float
```

Comment:
Not really, I'm kind of inconsistent with all these currently. I actually prefer the explicit string versions, this must have happened in a weak moment.